### PR TITLE
Added copy and move constructors for AdaptiveMaxPooling

### PR DESCRIPTION
**Pull request by ahmedr2001**

### CODE REVIEW

**Review by dnabanita7 — COMMENTED** [review]
Thanks for the PR! You can take help from https://github.com/mlpack/mlpack/pull/3128. Also, adding a test example so that it will be easier to debug if there's an issue with the implementation. You can take help from https://github.com/mlpack/mlpack/pull/3136/files

File: src/mlpack/methods/ann/layer/adaptive_max_pooling.hpp [hunk 1]
```diff
@@ -51,6 +51,12 @@ class AdaptiveMaxPooling
    * @param outputShape A two-value tuple indicating width and height of the output.
    */
   AdaptiveMaxPooling(const std::tuple<size_t, size_t>& outputShape);
+    
+  //Copy constructor
+  AdaptiveMaxPooling(const AdaptiveMaxPooling& pooling);
+    
+  //Move constructor
+  AdaptiveMaxPooling(AdaptiveMaxPooling&& pooling);
```
Comment (dnabanita7) [inline src/mlpack/methods/ann/layer/adaptive_max_pooling.hpp:59]: You can create an assignment operator for both copy and move constructors as you can see in other examples.

**Comment by eshaanagarwal** [comment]
Hi ! Thanks for the PR. 
But as per this [comment](https://github.com/mlpack/mlpack/issues/2625#issuecomment-1027376686), i am not sure whether we can move forward with this one.

**Comment by ahmedr2001** [comment]
I will try another issue instead, thanks a lot for your replies.

**Comment by rcurtin** [comment]
Since #2777 was merged, I think that this is not applicable anymore.  Sorry that this did not end up getting merged.  But thank you nonetheless for putting in the effort!

**Pull request closed without merging**